hrm-domain/packages/hrm-search-config/generateElasticsearchQuery.ts

```diff
@@ -211,34 +211,28 @@ const generateQueriesFromId = <TDoc extends DocumentType>({
 }): QueryDslQueryContainer[] => {
   const terms = searchTerm.split(' ');
 
-  const queries = terms
-    .map(term => {
-      // Ignore terms that are not entirely a number, as that breaks term queries against integer fields
-      if (Number.isNaN(Number(term)) || !Number.isInteger(Number(term))) {
-        return null;
-      }
-
-      const parsed = Number.parseInt(term, 10);
-
-      // Ignore numbers that are greater than maximum supported int
-      if (parsed > MAX_INT) {
-        return null;
-      }
-
-      return generateESQuery(
-        queryWrapper({
-          documentType,
-          type: 'term',
-          term: parsed,
-          boost: boostFactor * BOOST_FACTORS.id,
-          field: 'id' as any, // typecast to conform TS, only valid parameters should be accept
-          parentPath,
-        }),
-      );
-    })
-    .filter(q => q !== null) as QueryDslQueryContainer[]; // this typecast is awful but Array.filter does not infers that nulls are being removed
+  if (terms.length > 1) {
```

Comment:
This feels messy, generating a special query if the input is only a number.

I would much rather have a way of explicitly searching by ID in the UI and enforce numeric input there if we want to separate them, which would be another story. Otherwise just have the ID as part of fields searched on by the simple search. I don't think we should be trying to second guess whether what the user is actually searching for when they type a number. After all, an ID match is only ever going to account for 1 search result in the overall set returned, it's not the end of the world if that wasn't what they  were searching on

Reply:
Yes but that requires more changes vs what I'm trying to do here, which is to stop a paging error.

I'm curious: if there were major differences on what I pushed initially vs what would have been ideal from an ElasticSearch implementation, for both, queries and mappings - why weren't those mentioned in the early PRs? 
Trying to squeeze your suggestions as part of a quick bug fix is not really desirable IMO. Can we get a new ticket to address those changes?

Reply:
@stephenhand After experimenting a bit:
A query like
```
{
  "simple_query_string": {
    "fields": [
      "id^10", "content"
    ],
    "query": "\"2432 123\"",
    "flags": "FUZZY|NEAR|PHRASE|WHITESPACE",
    "boost": 2
  }
}
```
results in yet another error.
```
"type": "query_shard_exception",
"reason": "failed to create query: Can only use fuzzy queries on keyword and text fields - not on [id] which is of type [integer]",
```
This is true for any operator that we include, and also happens if the search term contains non-numerical characters.

> Otherwise just have the ID as part of fields searched on by the simple search.

I understand your suggestion, and I agree. But this is currently not possible, as it will cause more 5xx down the road.
We will need to change the mapping for this to be possible.

Reply:
Fair enough, I was hoping we could get away with it as is, that it might do some automatic casting, but your PR is fine to get the fix out in light of this

Can we create a ticket for the fix that requires the remap. It might be good to do that before we roll out the feature to all helplines